scripts/Run-Self-Hosted-Runner-Test.ps1

```diff
@@ -0,0 +1,261 @@
+
+# While at it, enable EULA for all SysInternals tools.
+REG ADD HKCU\Software\Sysinternals /v EulaAccepted /t REG_DWORD /d 1 /f | Out-Null
+
+# The following 'Set-ItemProperty' command enables a full memory dump.
+# Not enabled yet as it needs a VM with an explicitly created page file of at least (physical_memory + 1MB) in size.
+# The default value of the 'CrashDumpEnabled' key is 7 ('automatic' sizing of dump file size (system determined)).
+# https://learn.microsoft.com/en-us/troubleshoot/windows-server/performance/memory-dump-file-options
+#
+# Set-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\CrashControl' -Name 'CrashDumpEnabled' -Value 1
+#
+if ($VerbosePreference -eq 'Continue') {
+    # Dump current kernel mode dump settings.
+    Write-Log "`n"
+    Write-Log "Current kernel dump configuration:`n"
+    $KernelDumpInfo = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\CrashControl'
+    $Lines = $KernelDumpInfo -split '; '
+    foreach($Line in $Lines) {
+        Write-Log "`t$Line"
+    }
+    Write-Log "`n"
+}
+
+# Get the available free space before test start (useful in investigating dump file creation failures)
+$BeforeTestFreeSpaceGB = GetDriveFreeSpaceGB -DriveSpecification $Env:SystemDrive
+Write-Log "Available System disk space (Before test start): $BeforeTestFreeSpaceGB GB"
+
+# Start the test process using the provided command and arguments.
+$FullTestCommandSpec = Join-Path $Pwd $TestCommand
+Write-Log "`n`n"
+Write-Log "Staring Test command: $FullTestCommandSpec $TestArguments"
+Write-Log "Test hang timeout: $TestHangTimeout (seconds)"
+Write-Log "`n"
+
+# Create a temporary file for standard output and error.
+$TestTempOutFile = [System.IO.Path]::GetTempFileName()
+
+# Form the complete command line with output redirection for cmd.exe.
+$TestCommandLine = "$FullTestCommandSpec $($TestArguments) > $TestTempOutFile 2>&1"
+
+# Start the test process and wait for it to exit or timeout.
+$TestProcess = Start-Process -FilePath "cmd.exe" -ArgumentList "/c $TestCommandLine" -PassThru -NoNewWindow
+
+# Cache the process handle. This has a side-effect that enables access to the process's exit code after termination.
+# (https://stackoverflow.com/a/23797762)
+$ProcessHandle = $TestProcess.Handle
+
+$WaitResult = $TestProcess.WaitForExit($TestHangTimeout * 1000)
+if (-not $WaitResult) {
+    Write-Log "`n"
+    Write-Log "*** ERROR *** Test execution hang timeout ($TestHangTimeout seconds) expired.`n"
+
+    # First, generate a user mode dump.
+    $UserModeDumpFileName = "$($TestCommand)_$(Get-Date -Format 'yyyy-MM-dd_HH-mm-ss').dmp"
+    $UserModeDumpFilePath = Join-Path $UserModeDumpFolder $UserModeDumpFileName
+
+    if ($VerbosePreference -eq 'Continue') {
+        Write-Log "User mode Dumpfile name: $UserModeDumpFileName"
+        Write-Log "User mode Dumpfile Path: $UserModeDumpFilePath"
+    }
+
+    # Get the available free space at this point in case the test creates files of it's own.
```

Comment:
nit: "files of its own" or simpler "its own files"

Reply:
fixed.